edb/pgsql/common.py

```diff
@@ -609,3 +610,180 @@ def get_sql_value_function_op(op: pgast.SQLValueFunctionOP) -> str:
+    'enable_seqscan',
+    'enable_sort',
+    'enable_tidscan',
+    'escape_string_warning',
+    'event_triggers',
+    'exit_on_error',
+    'fsync',
+    'full_page_writes',
+    'geqo',
+    'gss_accept_delegation',
+    'hot_standby',
+    'hot_standby_feedback',
+    'huge_pages',
+    'huge_pages_status',
+    'icu_validation_level',
+    'ignore_checksum_failure',
+    'ignore_invalid_pages',
+    'ignore_system_indexes',
+    'in_hot_standby',
+    'integer_datetimes',
+    'intervalstyle',
+    'jit',
+    'jit_debugging_support',
+    'jit_dump_bitcode',
+    'jit_expressions',
+    'jit_profiling_support',
+    'jit_tuple_deforming',
+    'krb_caseins_users',
+    'lo_compat_privileges',
+    'log_checkpoints',
+    'log_connections',
+    'log_disconnections',
+    'log_duration',
+    'log_error_verbosity',
+    'log_executor_stats',
+    'log_hostname',
+    'log_lock_waits',
+    'log_min_error_statement',
+    'log_min_messages',
+    'log_parser_stats',
+    'log_planner_stats',
+    'log_recovery_conflict_waits',
+    'log_replication_commands',
+    'log_statement',
+    'log_statement_stats',
+    'log_truncate_on_rotation',
+    'logging_collector',
+    'parallel_leader_participation',
+    'password_encryption',
+    'plan_cache_mode',
+    'quote_all_identifiers',
+    'recovery_init_sync_method',
+    'recovery_prefetch',
+    'recovery_target_action',
+    'recovery_target_inclusive',
+    'remove_temp_files_after_crash',
+    'restart_after_crash',
+    'row_security',
+    'send_abort_for_crash',
+    'send_abort_for_kill',
+    'session_replication_role',
+    'shared_memory_type',
+    'ssl',
+    'ssl_max_protocol_version',
+    'ssl_min_protocol_version',
+    'ssl_passphrase_command_supports_reload',
+    'ssl_prefer_server_ciphers',
+    'standard_conforming_strings',
+    'stats_fetch_consistency',
+    'summarize_wal',
+    'sync_replication_slots',
+    'synchronize_seqscans',
+    'synchronous_commit',
+    'syslog_facility',
+    'syslog_sequence_numbers',
+    'syslog_split_messages',
+    'trace_connection_negotiation',
+    'trace_notify',
+    'trace_sort',
+    'track_activities',
+    'track_commit_timestamp',
+    'track_counts',
+    'track_functions',
+    'track_io_timing',
+    'track_wal_io_timing',
+    'transaction_deferrable',
+    'transaction_isolation',
+    'transaction_read_only',
+    'transform_null_equals',
+    'update_process_title',
+    'wal_compression',
+    'wal_init_zero',
+    'wal_level',
+    'wal_log_hints',
+    'wal_receiver_create_temp_slot',
+    'wal_recycle',
+    'wal_sync_method',
+    'xmlbinary',
+    'xmloption',
+    'zero_damaged_pages',
+}
+
+
+def setting_to_sql(name, setting):
+    is_enum = name.lower() in ENUM_SETTINGS
+
+    assert typeutils.is_container(setting)
+    return ', '.join(setting_val_to_sql(v, is_enum) for v in setting)
+
+
+def setting_val_to_sql(val: str | int | float, is_enum: bool):
+    if isinstance(val, str):
+        if is_enum:
```

Comment:
I'm not sure why this is needed? Quoted enum config vars seem to work fine?

Reply:
@aljazerzen

Reply:
https://github.com/geldata/gel/pull/7745

Reply:
I *think* the original problem was that I was trying to set a setting that is expected to be an int, but our machinery was converting it to string.

So I refactored the machinery so that python always passes around `string | int | float` and then correctly serializes to sql.

Reply:
Well, the `to_sql` is a misnomer here then, because those values don't actually end up in SQL and are exposed at protocol level.